### ISSUE TITLE
`devenv container` fails to install pip requirements

### ISSUE THREAD

**Issue by edulix** [cachix/devenv#751]
**Describe the bug**

`devenv container` fails to install venv requirements, resulting of them not available in the container. It looks like 

**To reproduce**

To reproduce you can use this repository: https://github.com/sequentech/test-devenv 

Code (green button) -> Codespaces -> "Create Codespace on main" and it should work directly from the browser. The README explains the issue(s) and how to reproduce it.

The crux of the matter seems to be an issue with pip: `pip: cannot execute: required file not found`:

```
devenv container shell --docker-run
/nix/store/h2mp6lz6jjih6cdr17wfa916vm6nibiv-image-shell.json

Copying container /nix/store/h2mp6lz6jjih6cdr17wfa916vm6nibiv-image-shell.json to docker-daemon:shell:latest

Getting image source signatures
Copying blob 3789085753e8 done  
Copying config 1787d00664 done  
Writing manifest to image destination
/nix/store/g0nnvrmhndbzcqa93w7dhqqfygq8g83s-init-venv.sh: line 21: //.devenv/state/venv/bin/pip: cannot execute: required file not found
✨ devenv 0.6.2 is out of date. Please update to 0.6.3: https://devenv.sh/getting-started/#installation
git version 2.41.0
```

For some reason, pip cannot run

**Version** 0.6.2

**Comment by edulix** [comment]
I've tried to convert the project to poetry just to see if that works. Still doesn't work although since the init script is different, it doesn't give the same error log. But I found that if I add the correct path to the python path it seems to be fixed - but only if the command is executed from `enterShell`, not from the container's `startupCommand`.

I updated the repo's readme with poetry and with info about that, also please look at the [`devenv.nix`](https://github.com/sequentech/test-devenv/blob/bbce9de69e6f128c4acd006bf503af9284dd92eb/devenv.nix)

**Comment by thamenato** [comment]
Bringing the conversation from Discord to this issue -

I also have the same error as @edulix but I started my project already using poetry and I've noticed an odd behavior -

When using `devenv shell` things work as expected

```
❯ devenv shell
Building shell ...
Entering shell ...

(devenv) (sum-array-py3.11) thamenato@thales-meerkat:~/kairos/poc-argo-workflows/services/sum-array$ python --version
Python 3.11.4
```

but when trying to use processes it gives me a different python version

```
❯ devenv up
Building shell ...
09:13:38 system     | mypython.1 started (pid=20104)
09:13:38 mypython.1 | Python 3.10.12
09:13:38 system     | mypython.1 stopped (rc=0)
```

and the last one if the same problem mentioned on this issue

```
❯ devenv container mypython --docker-run
/nix/store/207wfq2pmbx6dcpkjhsr93g8ibldc0cg-image-my-python.json

Copying container /nix/store/207wfq2pmbx6dcpkjhsr93g8ibldc0cg-image-my-python.json to docker-daemon:my-python:latest

Getting image source signatures
Copying blob 8c3bf258c5d2 done  
Copying config e52f8d36ee done  
Writing manifest to image destination
Installing dependencies from lock file
/nix/store/ll2v7k63pf7qs40q3s02sqf5qvj8qx1p-entrypoint: line 7: exec: python --version: not found
```

When I run docker using the image that was just created but add the cmd in front of it

```
❯ docker run --rm -it my-python python --version
Installing dependencies from lock file
Python 3.11.4
```

**Comment by mcdonc** [comment]
Can confirm original error on greenfield project.

**Comment by domenkozar** [comment]
I think this should be fixed #745
